src/test/java/picard/analysis/CollectRnaSeqMetricsTest.java

```diff
@@ -212,11 +217,86 @@ else if (metrics.LIBRARY.equals("bar")) {
                 Assert.assertEquals(metrics.CORRECT_STRAND_READS, 0);
                 Assert.assertEquals(metrics.INCORRECT_STRAND_READS, 2);
                 Assert.assertEquals(metrics.IGNORED_READS, 1);
-
+                Assert.assertEquals(metrics.PCT_FIRST_READ_TRANSCRIPTION_STRAND_READS, 1.0);
+                Assert.assertEquals(metrics.PCT_SECOND_READ_TRANSCRIPTION_STRAND_READS, 0.0);
             }
         }
     }
 
+    @Test
+    public void testTranscriptionStrandMetrics() throws Exception {
+        final String sequence = "chr1";
+        final String ignoredSequence = "chrM";
+
+        // Create some alignments that hit the ribosomal sequence, various parts of the gene, and intergenic.
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+        // Set seed so that strandedness is consistent among runs.
+        builder.setRandomSeed(0);
+        final int sequenceIndex = builder.getHeader().getSequenceIndex(sequence);
+
+        // Reads that start *before* the gene: not counted
+        builder.addPair("pair_prior_1", sequenceIndex, 45, 50, false, false, "50M", "50M", true, false, -1);
```

Comment:
could you please add:

1. secondary alignments
2. supplementary alignments
3. records that do not have MC tag
4. duplicate reads.
5. ignored sequence reads
6. non-PF reads